secure_message/services/party_service.py

```diff
@@ -34,6 +34,36 @@ def get_users_details(self, uuids):
         """Retrieves the user details from the party service"""
         return self._get_user_details_from_party_service(uuids)
 
+    @staticmethod
+    def does_user_have_claim(user_id, bus_id, survey_id):
+        """Returns true of the user id has a claim on the bus_id, survey_id combination
+         False if not.
+
+         Rather than inspect the respondent data it defers to Party Service for the precise logic
+         to determine what constitutes a valid claim so as to maintain separation of concerns and
+         avoid future maintenance issues.
+        """
+        params = {"respondent_id": user_id, "bus_id": bus_id, "survey_id": survey_id}
+
+        response = requests.get(f"{current_app.config['RAS_PARTY_SERVICE']}party-api/v1/respondents/claim",
+                                auth=current_app.config['BASIC_AUTH'], verify=False, params=urlencode(params))
+        try:
+            response.raise_for_status()
+        except HTTPError:
+            logger.exception('Failed to validate claim data with party service',
+                             status_code=response.status_code,
+                             text=response.text,
+                             respondent_id=user_id,
+                             bus_id=bus_id,
+                             survey_id=survey_id)
+            return []
```

Comment:
May be my lack of Python showing but it seems odd to return `[]` here rather than `False`

Reply:
Agreed, the comment said True and False return values and the code below returns a true.  All sorts of messes happen when you have functions that return different types!

Reply:
Agreed , modifying